lib/hip/hipUtils.hpp

```diff
@@ -0,0 +1,30 @@
+/**
+ * @file
+ * @brief Helper function for using HIP
+ *  - CHECK_HIP_ERROR
+ *  - CHECK_HIP_ERROR_NON_OO
+ */
+
+#ifndef HIP_UTILS_H
+#define HIP_UTILS_H
+
+#include "errors.h"
+#include "kotekanLogging.hpp"
+
+#include <assert.h>
+
+// Convenience function for checking HIP runtime API results
+// can be wrapped around any runtime API call. No-op in release builds.
+#define CHECK_HIP_ERROR(result)                                                                    \
+    if (result != hipSuccess) {                                                                    \
+        FATAL_ERROR_NON_OO("Error at {:s}:{:d}; Error type: {:s}", __FILE__, __LINE__,             \
+                           hipGetErrorString(result));                                             \
+    }
```

Comment:
Should I expect this macro to be identical to the NON_OO one?

Reply:
It's almost the same.  The key difference is the `NON_OO` uses the global config log level (instead of the per-stage overridable log_level).  However, in cases like this where it might not be called from a function or class that doesn't inherit from `kotekanLogging` we have to use this version.